rotkehlchen/api/rest.py

```diff
@@ -4675,15 +4678,35 @@ def export_history_events(
             )
 
         if len(history_events) == 0:
-            return api_response(
-                wrap_in_fail_result('No history processed in order to perform an export'),
+            return wrap_in_fail_result(
+                message='No history processed in order to perform an export',
                 status_code=HTTPStatus.CONFLICT,
             )
 
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            settings = self.rotkehlchen.get_settings(cursor)
+            currency = settings.main_currency.resolve_to_asset_with_oracles()
+
         serialized_history_events = []
         headers: dict[str, None] = {}
         for event in history_events:
-            serialized_event = event.serialize_for_csv()
+            if event.balance.usd_value != ZERO and currency != A_USD:
+                try:
+                    price = self.get_rate_in_profit_currency(
+                        asset=event.asset,
+                        timestamp=ts_ms_to_sec(event.timestamp),
+                        currency=currency,
+                    )
+                except (PriceQueryUnsupportedAsset, RemoteError):
+                    price = ZERO_PRICE
+                except NoPriceForGivenTimestamp as e:
+                    # In the case of NoPriceForGivenTimestamp when we got rate limited we let
+                    # it propagate so the user can take action after the report is made
+                    if e.rate_limited is True:
+                        raise
+                    price = ZERO_PRICE
+                event.balance.usd_value = event.balance.amount * price
+            serialized_event = event.serialize_for_csv(currency)
```

Comment:
instead of the approach of modifying the event.balance data what I would do is to provice an argument for this function `price_in_native_currency` and what you do is querying the pricce in the native currency if it is != USD and provide it as argument.

Since you don't have the `price_in_usd` you can set `price_in_native_currency` with default to None and check 

```
if price_in_native_currency is not None:
     fiat_value = price_in_native_currency * event.balance.amount
else:
     fiat_value = event.balance.value
```

does it make sense? Also properly document it